# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufSurface.h
# Commit: c3d14f2b234a
# Full Hash: c3d14f2b234a7494edf3231113af52f5fc2d2355
# Author: stransky <stransky@redhat.com>
# Date: 2022-01-25 16:37:35
# Regressor Bug: 1724385
# File Overlap Count: 1
# Description:
#   Bug 1724385 [Linux] Try to create EGLImage over decoded VA-API video frame r=alwu,emilio,media-playback-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D135460
# ==============================================================================

diff -r 5834c469b533 -r c3d14f2b234a widget/gtk/DMABufSurface.h
--- a/widget/gtk/DMABufSurface.h	Tue Jan 25 07:03:13 2022 +0000
+++ b/widget/gtk/DMABufSurface.h	Tue Jan 25 07:08:02 2022 +0000
@@ -308,6 +308,8 @@
   bool UpdateYUVData(void** aPixelData, int* aLineSizes);
   bool UpdateYUVData(const VADRMPRIMESurfaceDescriptor& aDesc);
 
+  bool VerifyTextureCreation();
+
  private:
   ~DMABufSurfaceYUV();
 
@@ -324,6 +326,9 @@
   void CloseFileDescriptorForPlane(const mozilla::MutexAutoLock& aProofOfLock,
                                    int aPlane, bool aForceClose);
 
+  bool CreateEGLImage(mozilla::gl::GLContext* aGLContext, int aPlane);
+  void ReleaseEGLImages(mozilla::gl::GLContext* aGLContext);
+
   int mWidth[DMABUF_BUFFER_PLANES];
   int mHeight[DMABUF_BUFFER_PLANES];
   EGLImageKHR mEGLImage[DMABUF_BUFFER_PLANES];
